item.rb

```diff
@@ -38,9 +38,10 @@ def move_to_archive
   end
 
   def parse_date(date_str)
-    Date.parse(date_str)
+    Date.parse(date_str.to_s)
   rescue ArgumentError, TypeError
-    raise ArgumentError, 'Invalid publish date format'
+    raise ArgumentError, 'Invalid publish date format. Please use the format YYYY-MM-DD.'
   end
```

Comment:
The current code raises two separate ArgumentError exceptions with different error messages. It's generally better to provide a single, informative error message that covers both cases. Additionally, it's a good practice to include the original exception's message in the error message to provide more context about the specific issue.